### ISSUE TITLE
Option not to display Asset Tag value appended to Device Name

### ISSUE THREAD

**Issue by MattDRogers** [netbox-community/netbox#17425]
### NetBox version

v4.1.0

### Feature type

Change to existing functionality

### Proposed functionality

Either do _not_ append the Asset Tag to the Device Name, or make it an _option_.

**What it looks like now:**
![Screenshot 2024-09-09 092416](https://github.com/user-attachments/assets/934dba5e-3f29-4397-a097-661791d98c3d)
![Screenshot 2024-09-09 092549](https://github.com/user-attachments/assets/3e8a6469-39e9-4f06-a0cd-6f496b111fc0)
![Screenshot 2024-09-09 092744](https://github.com/user-attachments/assets/171a84cb-6502-4c80-b168-e217bfcca22d)

**What I'd like the option to do:**
![Screenshot 2024-09-09 092416a](https://github.com/user-attachments/assets/44cea39a-af66-472f-902e-0758094c7459)
![Screenshot 2024-09-09 092549a](https://github.com/user-attachments/assets/b5c75507-2978-4965-8284-d7f76a4d6347)
![Screenshot 2024-09-09 092744a](https://github.com/user-attachments/assets/c6a11d8c-7438-4f80-86d5-9361589438e2)

### Use case

For display purposes, appending the Asset Tag to the Device Name uses up a lot of space--especially when one is looking at a list of cable connections.  It makes for a more cluttered view (though I suppose this is subjective). Having the option to disable the parenthetical Asset Tag appearing next to the Device Name would streamline things visually, make tables display in a more easy-to-read fashion, and eliminate needless redundancy of information.

### Database changes

I don't believe any database changes should be necessary, as these are two separately defined fields being displayed a certain way in the Web UI.

### External dependencies

No external dependencies.

**Comment by goteamkor** [comment]
I agree. Asset tag seems to be the only other field that gets added to a name if you have it filled in.
It was so cumbersome for our users we don't use the built-in asset tag field, instead opting to create a custom field for asset tag just so it won't get appended to the name.

**Comment by jose-agudo** [comment]
Asset tag takes so much screen space and most of the time is irrelevant in most of the places or queries that include the device-name. Please consider the FR.

**Comment by jeremystretch** [comment]
This will probably involve removing the asset tag from the string returned by the model's `__str__()` method. However, IMO it make sense to still include the asset tag as part of the dropdown choices.

**Comment by alisenkov** [comment]
yes, we also need not showing asset tag in hostname

**Comment by alehaa** [comment]
Do we need a user preference for this or could it be a configuration setting? If the latter, I’m willing to volunteer to create a PR.

**Comment by pheus** [comment]
Thanks, @alehaa! A global configuration setting seems like the right first step. If there’s interest in per‑user control later, we can track that as a separate feature request.
Are you still up for opening a PR?
